lib/stbcbckt.gi

```diff
@@ -2955,24 +2955,41 @@ end );
 InstallMethod( Intersection2, "perm groups", IsIdenticalObj,
   [ IsPermGroup, IsPermGroup ], 0,
 function( G, H )
-local   Omega,  P,  rbase,  L,mg,mh,i;
-    
+local   Omega, P, rbase, L, mg, mh, IsFinished, mg_minus_mh, mh_minus_mg;
+
     if IsIdenticalObj( G, H )  then
       return G;
     fi;
-    
-    # align the acting domains
-    mg:=MovedPoints(G);
-    mh:=MovedPoints(H);
-    Omega := Intersection(mg,mh);
 
-    # no two points moved in common?
-    if Length(Omega)<=1 then
-      return TrivialSubgroup(Parent(G));
-    fi;
+    # iterate taking stabilizer until G and H have the same set of moved points
+    while(true)
+    do
+      # align the acting domains
+      mg:=MovedPoints(G);
+      mh:=MovedPoints(H);
+      Omega := Intersection(mg,mh);
+
+      # no two points moved in common?
+      if Length(Omega)<=1 then
+        return Group(());
```

Comment:
I agree the "Parent" didn't make a lot of sense here, but I would instead write `TrivialSubgroup(G)`. The reason to do `TrivialSubgroup`, instead of `Group(())`, is that GAP knows the result of `TrivialSubgroup` has size 1 (as the method sets it at construction time), while `Group(())` doesn't set that (as GAP doesn't special-case making an identity group)

Reply:
Actually, it does:
```
gap> HasIsTrivial(Group(()));
true
```
The more salient difference is the set of generators:
```
gap> G:=Group((1,2));; H:=Group((1,2,3));;
gap> GeneratorsOfGroup(Intersection(G,H));
[  ]
gap> GeneratorsOfGroup(TrivialSubgroup(G));
[  ]
gap> GeneratorsOfGroup(Group(()));
[ () ]
```

But note that using `G` as parent is still arbitrary (though harmless). All in all, I would recommend this:
```suggestion
        return TrivialGroup(IsPermGroup);
```
But using `TrivialSubgroup(G)` would also be fine.